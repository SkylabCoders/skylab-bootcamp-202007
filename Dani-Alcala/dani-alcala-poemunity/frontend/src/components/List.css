.list__container {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
}

.list__presentation {
  display: none;
  margin-bottom: 35px;
}

.poems__content {
  overflow: hidden;
  text-overflow: ellipsis;
  display: -webkit-box;
  /* number of lines to show */
  -webkit-line-clamp: 9;
  -webkit-box-orient: vertical;
}

.poems__read-more {
  margin-bottom: 20px;
  font-size: 14px;
  font-weight: bold;
  color: red;
  text-align: center;
}

.list__intro {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  margin-bottom: 10px;
}

.list__sort {
  margin-right: 5%;
  margin-bottom: 4px;
}

.list__search {
  margin-right: 5%;
}

@media screen and (min-width: 700px) {
  .list__container {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: vertical;
    -webkit-box-direction: normal;
        -ms-flex-direction: column;
            flex-direction: column;
    -webkit-box-pack: center;
        -ms-flex-pack: center;
            justify-content: center;
  }
  .list__intro {
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-orient: horizontal;
    -webkit-box-direction: normal;
        -ms-flex-direction: row;
            flex-direction: row;
    -webkit-box-align: end;
        -ms-flex-align: end;
            align-items: flex-end;
    -ms-flex-pack: distribute;
        justify-content: space-around;
  }
  .list__presentation {
    display: block;
    font-size: 20px;
    margin-left: 5%;
  }
}
/*# sourceMappingURL=List.css.map */