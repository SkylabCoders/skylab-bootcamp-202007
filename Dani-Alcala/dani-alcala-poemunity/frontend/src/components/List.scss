.list__container {
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.list__presentation {
    display: none;
    margin-bottom: 35px;
}

.poems__content {
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    /* number of lines to show */
    -webkit-line-clamp: 9;
    -webkit-box-orient: vertical;
}

.poems__read-more {
    margin-bottom: 20px;
    font-size: 14px;
    font-weight: bold;
    color: red;
    text-align: center;
}

.list__intro {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin-bottom: 10px;
}

.list__sort {
    margin-right: 5%;
    margin-bottom: 4px;
}

.list__search {
    margin-right: 5%;
}

@media screen and (min-width: 700px) {
    .list__container {
        display: flex;
        flex-direction: column;
        justify-content: center;
    }

    .list__intro {
        display: flex;
        flex-direction: row;
        align-items: flex-end;
        justify-content: space-around;
    }

    .list__presentation {
        display: block;
        font-size: 20px;
        margin-left: 5%;
    }
}