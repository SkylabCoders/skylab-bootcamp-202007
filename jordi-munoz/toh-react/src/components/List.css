body {
    margin: auto;
    font-family: Arial, Helvetica, sans-serif;
    margin-left: 50px;
  }
.containerList {
    margin-top: 40px;
    display: flex;
    flex-direction: column;
  }
  .containerList p {
    color: rgb(103, 103, 139);
    background-color: rgb(204, 204, 204);
    margin: 4px;
    padding: 10px 50px;
    border-radius: 5px;
    border: none;
    font-size: 14px;
    font-weight: bold;
    width: 80px;
    text-align: center;
  }