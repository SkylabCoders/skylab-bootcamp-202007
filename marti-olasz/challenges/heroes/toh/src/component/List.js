import React from 'react';

import { NavLink, Route } from 'react-router-dom';

function List({ list }) {
	return list.map((hero) => {
		return (
			<NavLink to={'/hero/' + hero.id} key={hero.id}>
				<p>{hero.id + '  |  ' + hero.name}</p>
			</NavLink>
		);
	});
}

export default List;
