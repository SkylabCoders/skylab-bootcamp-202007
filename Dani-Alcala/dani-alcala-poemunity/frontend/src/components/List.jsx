import React, { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import {
  loadPoems,
  deletePoem,
  likePoem,
  sortByCriteria,
} from "../actions/poemActions";
import poemStore from "../stores/poemStore";
import "./List.scss";
import "./Detail.scss";
import "../App.scss";
import { TextField } from "@material-ui/core";
import SearchIcon from "@material-ui/icons/Search";
import HighlightOffSharpIcon from "@material-ui/icons/HighlightOffSharp";
import SubjectSharpIcon from "@material-ui/icons/SubjectSharp";
import { useAuth0 } from "@auth0/auth0-react";
import CircularProgress from "./CircularIndeterminate";


function List(props) {
  const { user, isAuthenticated, isLoading } = useAuth0();
  const genre = props.match.params.genre;
  const [poems, setPoems] = useState(poemStore.getPoems(genre));
  const [sort, setSort] = useState("Likes");
  const [filter, setFilter] = useState("");

  useEffect(() => {
    poemStore.addChangeListener(onChange);
    if (poems.length === 0) loadPoems();
    else {
      onChange();
    }
    return () => poemStore.removeChangeListener(onChange);
  }, [poems.length, genre, sort]);

  function onChange() {
    setPoems(poemStore.getPoems(genre));
  }

  function onDelete(event, poemId) {
    event.preventDefault();
    deletePoem(poemId);
  }

  function onLike(event, poemId, userId) {
    event.preventDefault();
    likePoem(poemId, userId);
  }

  const LIKE = "Like";
  const LIKES = "Likes";
  const READ_MORE = "Leer más";
  const ADMIN = "google-oauth2|102774395820207939159";

  const handleSearchChange = (event) => {
    setFilter(event.target.value);
  };

  function onFieldChange(value, setValue) {
    setValue(value);
  }

  if (isLoading) {
    return <CircularProgress />;
  }

  return (
    <>
      <div className="list__container">
        <div className="list__intro">
          {!genre && (
            <p className="list__presentation">¡Tu red social de poesía!</p>
          )}
          {genre && (
            <p className="list__presentation">
              Categoría: {genre.toUpperCase()}
            </p>
          )}
          <div className="separator"></div>
          <form className="list__sort">
            <label>
              {`Ordenar poemas por: `}
              <select
                type="submit"
                id="sort"
                name="sort"
                onChange={(event) => {
                  onFieldChange(event.target.value, setSort);
                  sortByCriteria(event.target.value);
                }}
              >
                <option value="likes">Likes</option>
                <option value="fecha">Fecha</option>
                <option value="aleatorio">Aleatorio</option>
                <option value="titulo">Título</option>
              </select>
            </label>
          </form>
          <div className="list__search">
            <SearchIcon style={{ fontSize: 40, fill: "#4F5D73" }} />
            <TextField
              label="Buscar autor"
              InputLabelProps={{
                style: { color: "#4F5D73" },
              }}
              InputProps={{
                style: { color: "#4F5D73" },
              }}
              onChange={handleSearchChange}
            />
          </div>
        </div>

        {poems.map((poem) => (
          <main key={poem._id} className="poem__detail">
            {poem.author.includes(filter) && (
              <section className="poem__block">
                <section>
                  <Link to={`/detalle/${poem._id}`} className="poem__title">
                    {poem.title}
                  </Link>
                  <div className="poem__author-container">
                  <img className="poem__picture" src={poem.picture} />
                  <p className="poem__author">{poem.author}</p>
                  </div>
                  <div className="poem__date">{poem.date}</div>
                </section>
                <section>
                  <div className="poem__content poems__content">
                    {poem.poem}
                  </div>
                  <div className="poems__read-more">
                    <Link
                      to={`/detalle/${poem._id}`}
                      className="poems__read-more"
                    >
                      {READ_MORE}
                    </Link>
                  </div>
                </section>
                <section className="poem__footer">
                  {poem.likes.length === 1 && (
                    <div className="poem__likes">
                      {poem.likes.length} {LIKE}
                    </div>
                  )}
                  {poem.likes.length !== 1 && (
                    <div className="poem__likes">
                      {poem.likes.length} {LIKES}
                    </div>
                  )}
                  <div className="separator"></div>
                  {isAuthenticated &&
                    poem.author !== user.name &&
                    poem.likes.some((id) => id === user.sub) && (
                      <Link
                        className="poem__likes-icon"
                        onClick={(event) => onLike(event, poem._id, user.sub)}
                      ></Link>
                    )}
                  {isAuthenticated &&
                    poem.author !== user.name &&
                    !poem.likes.some((id) => id === user.sub) && (
                      <Link
                        className="poem__unlikes-icon"
                        onClick={(event) => onLike(event, poem._id, user.sub)}
                      ></Link>
                    )}
                  {isAuthenticated &&
                    (poem.author === user.name || user.sub === ADMIN) && (
                      <HighlightOffSharpIcon
                        className="poem__delete-icon"
                        style={{ fill: "red" }}
                        onClick={(event) => onDelete(event, poem._id)}
                      />
                    )}
                  <Link
                    to={`/detalle/${poem._id}`}
                    className="poem__comments-icon"
                  >
                    <SubjectSharpIcon style={{ fill: "#000" }} />
                  </Link>
                </section>
              </section>
            )}
          </main>
        ))}
      </div>
    </>
  );
}
export default List;
