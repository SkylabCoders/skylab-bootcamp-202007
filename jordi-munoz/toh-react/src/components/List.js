import React from 'react';
import './List.css';

function List({heroesList}) {
    return (
        <div className='containerList'>
            {heroesList}
        </div>
    )
}

export default List;