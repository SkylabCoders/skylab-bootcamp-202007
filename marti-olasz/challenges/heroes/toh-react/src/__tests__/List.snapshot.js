import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import List from '../components/HeroList';

describe('Lists snapshot', () => {
	const tree = renderer.create(
		<Router>
			<List />
		</Router>
	);
	it('Should match', () => {
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
