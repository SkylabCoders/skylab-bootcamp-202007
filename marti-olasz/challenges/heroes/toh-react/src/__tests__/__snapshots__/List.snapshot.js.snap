// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists snapshot Should match 1`] = `<ul />`;
